Docs improvements (#961)
anatawa12/AvatarOptimizer: .docs/content/docs/developers/distributing-prefabs/index.md

---
title: Distributing Prefabs
---

# Distributing Prefabs

Avatar Optimizer mey help you making your assets lightweight and optimized.
Here are some recommendations for how to distribute prefabs that depend on Avatar Optimizer.

## Guide your users to the official distribution of Avatar Optimizer {#guide-to-the-official-distribution}

> Since Avatar Optimizer is published under MIT License, it's allowed to redistribute it.

Reviewer: ```suggestion
Since Avatar Optimizer is published under MIT License, redistribution is allowed.
```

> However, I strongly recommend you to guide your users to the official distribution of Avatar Optimizer.

Reviewer: ```suggestion
However, I strongly recommend you to guide your users to the official distribution of Avatar Optimizer.\
```

> Your distribution of Avatar Optimizer may be outdated. 

Reviewer: ```suggestion
Unofficial distribution of Avatar Optimizer may be outdated. 
```

Author: 言いたいこととしてはあなたが配布すると古い可能性あるよねなのでUnofficialというよりYourだとおもう

Author: ```suggestion
Your unofficial distribution of Avatar Optimizer may be outdated. 
```

がちょうどいい？

Reviewer: 良いかも?

> If your distribution is outdated, it may have bugs that have already been fixed in the official distribution. 

Reviewer: ```suggestion
If users use outdated version, they may encounter bugs that have already been fixed in the official distribution. 
```

In addition, since Avatar Optimizer doesn't guarantee forward compatibility,

> if your user uses a newer version of Avatar Optimizer than your distribution, this will cause problems.

Reviewer: ```suggestion
if your users use a newer version of Avatar Optimizer than unofficial distribution, some problems will be caused.
```

Author: ```suggestion
if your users use a newer version of Avatar Optimizer than your unofficial distribution and downgrade it to outdated version, some problems will be caused.
```

There are recommend ways to guide your users to the official distribution of Avatar Optimizer.
1. Let users to install Avatar Optimizer by VPM if your asset is distributed with VPM.\

>    If your asset is distributed with VPM, declare Avatar Optimizer as a dependency of your asset in your `package.json`

Reviewer: ```suggestion
   Declare Avatar Optimizer as a dependency of your asset in your `package.json`
```

   and guide your users to add the Avatar Optimizer VPM repository to VCC.

>    [Here][add-repo] is the link to add the Avatar Optimizer repository to VCC.

Reviewer: ```suggestion
   [Here][add-repo] is the link to add the Avatar Optimizer Official repository to VCC.
```

   If you want, you can mirror the [Avatar Optimizer repository][repo] in your VPM repository.

>     When mirroring, mirror the VPM repository, not from the GitHub Releases.

Reviewer: ```suggestion
   When mirroring, mirror the VPM repository, not from the GitHub Releases.
```
なんかインデントずれてる?

2. Link to the official documentation of Avatar Optimizer.\

>    [This link][official-installation] is the official installation guide.

Reviewer: ```suggestion
   [This link][official-installation] is the official installation guide of Avatar Optimizer.
```

   The page will guide your users to install Avatar Optimizer in recommended way.

> 3. Link to [the booth page of Avatar Optimizer][booth-aao]. \

Reviewer: ```suggestion
3. Link to [the Booth page of Avatar Optimizer][booth-aao].\
```

>    The booth page is one of the official distribution page of Avatar Optimizer.

Reviewer: ```suggestion
   The Booth page is one of the official distribution page of Avatar Optimizer.
```

>    The booth item includes the latest version of VPAI installer unitypackage.

Reviewer: ```suggestion
   The Booth item includes the latest version of VPAI installer unitypackage.
```
```suggestion
   In the Booth page, the latest version of VPAI installer unitypackage is provided.
```

Author: memo: 後者

> 4. Include VPAI installer unitypackage and guide your users to import the unitypackage. \

Reviewer: ```suggestion
4. Include VPAI installer unitypackage and guide your users to import the unitypackage.\
```

   The VPAI installer is the tool to install VPM packages like Avatar Optimizer just by importing a unitypackage.

>    [Here][vpai] is the link to download the VPAI installer unitypackage for Avatar Optimizer 1.x.x

Reviewer: ```suggestion
   [Here][vpai] is the link to download the VPAI installer unitypackage for Avatar Optimizer 1.x.x .
```
試しに入れてみる
